test/parse.js

```diff
@@ -843,3 +843,52 @@ test('parse()', function (t) {
 
     t.end();
 });
+
+test('parse/stringify empty key WIP', function (t) {
+    t.test('parses an object with empty string key', function (st) {
+        st.deepEqual(qs.parse('=a', { allowEmptyKeys: true }), { '': 'a' });
+        st.deepEqual(qs.parse('=', { allowEmptyKeys: true }), { '': '' });
+
+        st.deepEqual(qs.parse('=&', { allowEmptyKeys: true }), { '': '' });
+        st.deepEqual(qs.parse('&=&', { allowEmptyKeys: true }), { '': '' });
+        st.deepEqual(qs.parse('&=', { allowEmptyKeys: true }), { '': '' });
+        st.deepEqual(qs.parse('&', { allowEmptyKeys: true }), { });
+        st.deepEqual(qs.parse('=&&&', { allowEmptyKeys: true }), { '': '' });
+
+        st.deepEqual(qs.parse('[deep]=a&[deep]=2', { allowEmptyKeys: true }), { '': { deep: ['a', '2'] } });
+        st.deepEqual(qs.parse('=a&=b', { allowEmptyKeys: true }), { '': ['a', 'b'] });
+        st.deepEqual(qs.parse('%5B0%5D=a&%5B1%5D=b', { allowEmptyKeys: true }), { '': ['a', 'b'] });
+        st.deepEqual(qs.parse('=a&foo=b', { allowEmptyKeys: true }), { '': 'a', foo: 'b' });
+        st.deepEqual(qs.parse('[]=a&[]=b& []=1', { allowEmptyKeys: true }), { '': ['a', 'b'], ' ': ['1'] });
+
+        st.deepEqual(qs.parse('=&a[]=b', { allowEmptyKeys: true }), { '': '', a: ['b'] });
+        st.deepEqual(qs.parse('=&a[]=b&a[1]=c', { allowEmptyKeys: true }), { '': '', a: ['b', 'c'] });
+        st.deepEqual(qs.parse('=&a[]=b&a[]=c&a[2]=d', { allowEmptyKeys: true }), { '': '', a: ['b', 'c', 'd'] });
+
+        st.deepEqual(qs.parse('a=b&a[]=c&=', { allowEmptyKeys: true }), { '': '', a: ['b', 'c'] });
+        st.deepEqual(qs.parse('a[]=b&a=c&=', { allowEmptyKeys: true }), { '': '', a: ['b', 'c'] });
+        st.deepEqual(qs.parse('a[0]=b&a=c&=', { allowEmptyKeys: true }), { '': '', a: ['b', 'c'] });
+        st.deepEqual(qs.parse('a=b&a[0]=c&=', { allowEmptyKeys: true }), { '': '', a: ['b', 'c'] });
+
+        st.deepEqual(qs.parse('a[]=b&a=c&=', { allowEmptyKeys: true }), { '': '', a: ['b', 'c'] });
+
+        st.end();
+
+    });
+
+    t.test('stringifies an object with empty string key', function (st) {
+        st.deepEqual(qs.stringify({ '': 'a' }), '=a');
+        st.deepEqual(qs.stringify({ '': ['a', 'b'] }), '%5B0%5D=a&%5B1%5D=b');
+        st.deepEqual(qs.stringify({ '': ['a', 'b'], a: [1, 2] }, { encode: false }), '[0]=a&[1]=b&a[0]=1&a[1]=2');
+
+        st.end();
+    });
+    t.test('edge case with object/arrays', function (st) {
+
+        st.deepEqual(qs.stringify({ '': { '': [2, 3] } }, { encode: false }), '[][0]=2&[][1]=3');
+
+        st.deepEqual(qs.parse('[][0]=2&[][1]=3', { allowEmptyKeys: true }), { '': { '': [2, 3] } });
```

Comment:
This fails case fails, but this is ambiguous string which is treated as `{ : [ '2', '3' ] }` and stringify <-> parse is not producing the same results